my-app/src/App.js

```diff
@@ -1,24 +1,80 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import './asset/scripts/i18n';
+import './asset/css/Product.css'
+import { Productdata } from './asset/scripts/productsdata';
+import Product from './components/Product';
 
 function App() {
+  const itemsPerPage = 8; // Number of items to display per page
+  const [currentPage, setCurrentPage] = useState(1);
+  const totalPages = Math.ceil(Productdata.length / itemsPerPage);
+  const currentProducts = Productdata.slice(
+    (currentPage - 1) * itemsPerPage,
+    currentPage * itemsPerPage
+  );
+  const handlePageChange = (newPage) => {
+    if (newPage >= 1 && newPage <= totalPages) {
+      setCurrentPage(newPage);
+    }
+  };
+  const pageNumbers = Array.from({ length: totalPages }, (_, index) => index + 1);
+  const { t, i18n } = useTranslation();
+  const changeLanguage = (lng) => {
+    i18n.changeLanguage(lng);
+  };
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
+    <div className="flex mr-10 border-t">
+      <div>
+      <h1>{t('category')}</h1>
+      <button onClick={() => changeLanguage('en')}>English</button>
+      <button onClick={() => changeLanguage('vi')}>Tiếng Việt</button>
+    </div>
```

Comment:
```suggestion
      <div>
        <h1>{t('category')}</h1>
        <button onClick={() => changeLanguage('en')}>English</button>
        <button onClick={() => changeLanguage('vi')}>Tiếng Việt</button>
      </div>
```
E đưa thành phần này và các xử lý liên quan vào components Category riêng nhé